@Ericson2314 

Looks promising, but I don't think I'm comfortable enough with Automake to make it work.

```
Makefile:1332: warning: overriding recipe for target 'b44ExpLogTable'
Makefile:751: warning: ignoring old recipe for target 'b44ExpLogTable'
Makefile:1338: warning: overriding recipe for target 'dwaLookups'
Makefile:755: warning: ignoring old recipe for target 'dwaLookups'
make[1]: *** No rule to make target 'b44ExpLogTable.c', needed by 'b44ExpLogTable.o'.  Stop.
make[1]: Leaving directory '/build/openexr-2.3.0/IlmImf'
make: *** [Makefile:502: all-recursive] Error 1
note: keeping build directory '/tmp/nix-build-openexr-2.3.0.drv-8'
builder for '/nix/store/fnwxk8b7jmgzalb2wp8v12m4w8gigd5z-openexr-2.3.0.drv' failed with exit code 2
error: build of '/nix/store/fnwxk8b7jmgzalb2wp8v12m4w8gigd5z-openexr-2.3.0.drv' failed
```

--- a/pkgs/development/libraries/openexr/default.nix
+++ b/pkgs/development/libraries/openexr/default.nix
@@ -1,4 +1,9 @@
-{ lib, stdenv, fetchurl, autoconf, automake, libtool, pkgconfig, zlib, ilmbase }:
+{ lib, stdenv, buildPackages, fetchurl, autoconf, automake, libtool, pkgconfig, zlib, ilmbase, }:
+
+let
+  # Doesn't really do anything when not crosscompiling
+  emulator = stdenv.hostPlatform.emulator buildPackages;
+in
 
 stdenv.mkDerivation rec {
   name = "openexr-${version}";
@@ -14,6 +19,22 @@ stdenv.mkDerivation rec {
   ];
 
   outputs = [ "bin" "dev" "out" "doc" ];
+
+  # Needed because there are some generated sources. Solution: just run them under QEMU.
+  postPatch = ''
+    for file in b44ExpLogTable dwaLookups
+    do
+      # Ecape for both sh and Automake
+      emu=${lib.escapeShellArg (lib.replaceStrings ["$"] ["$$"] emulator)}
+      before="./$file > $file.h"
+      after="$emu $before"
+      substituteInPlace IlmImf/Makefile.am \
+        --replace "$before" "$after"
+    done
+
+    # Make sure the patch succeeded
+    [[ $(grep "$emu" IlmImf/Makefile.am | wc -l) = 2 ]]
+  '';
 
   preConfigure = ''
     patchShebangs ./bootstrap

